shared/js/advanced_telemetry_helper.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * This module is a helper for recording telemetry data in Gecko’s telemetry
+ * system. Currently, only data from certified apps will be recorded.
+ *
+ * Usage Example:
+ *  var ath = new AdvancedTelemetryHelper();
+ *  // DO SOMETHING
+ *  // This creates and increments the counter on this metric called
+ *  // 'mycountmetric'
+ *  ath.count('mycountmetric'); // Count will be 1
+ *  ath.count('mycountmetric'); // Count is now 2
+ *  // DO SOMETHING
+ *  // This creates and logs a linear histogram called 'mylinearmetric' with
+ *  // value 100, minimum of 0, max of 1000, with 10 buckets.
+ *  ath.linear('mylinearmetric', 100, 0, 1000, 10);
```

Comment:
I'm wondering if it's cleaner to either 'initialize' the histogram (specify name, min, max, number of buckets) or to add a value, but not both. In other words, if the user has to explicitly initialize the histogram once, also adding a value at the same time might not be that helpful and may make the API less straightforward.